conans/model/version_range.py

```diff
@@ -140,6 +140,14 @@ def valid(self, version, conf_resolve_prepreleases):
                     return False
         return True
 
+    def __eq__(self, other):
+        if not isinstance(other, _ConditionSet):
```

Comment:
Is it a real possibility that a condition set is compared with something not a condition set? 
If it is a programmer protection, not logic flow, better replace it with an ``assert``

Reply:
I have simplified it, it was added by @AbrilRBS to make the tests pass, but removing the optimization also makes it work, and code is simpler